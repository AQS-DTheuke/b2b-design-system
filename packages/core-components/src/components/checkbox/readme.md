# b2b-checkbox



<!-- Auto Generated Below -->


## Properties

| Property             | Attribute  | Description                                                                                                                                 | Type      | Default     |
| -------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------- | --------- | ----------- |
| `checked`            | `checked`  | Whether or not the checkbox is checked. Default value is false and can be set to true if the checkbox should come pre-checked.              | `boolean` | `false`     |
| `disabled`           | `disabled` | Whether or not the checkbox is disabled. The default value is false.                                                                        | `boolean` | `false`     |
| `error`              | `error`    | The error message. It is undefined by default. If a string is passed in, it will render the checkbox with error styles.                     | `string`  | `undefined` |
| `hint`               | `hint`     | The hint text belonging to the checkbox. It is undefined by default. If an error is specified, it will be shown instead of the hint.        | `string`  | `undefined` |
| `invalid`            | `invalid`  | Whether or not the checkbox is rendered with error styles. Defaults to false.                                                               | `boolean` | `false`     |
| `label` _(required)_ | `label`    | The checkbox label. This attribute is required.                                                                                             | `string`  | `undefined` |
| `name`               | `name`     | The name of the checkbox. Per default it is undefined. Use this to programmatically group checkboxes together by giving them the same name. | `string`  | `undefined` |
| `required`           | `required` | Adds an asterisk at the end of the label to signify that the field is required.                                                             | `boolean` | `false`     |
| `value`              | `value`    | The value of the checkbox. This is not the same as the checked property. It is only used when the checkbox participates in a checkbox group | `any`     | `undefined` |


## Events

| Event        | Description                                                | Type                                    |
| ------------ | ---------------------------------------------------------- | --------------------------------------- |
| `b2b-blur`   | Emits whenever the checkbox loses focus.                   | `CustomEvent<FocusEvent>`               |
| `b2b-change` | Emits the checkbox value when it's checked status changes. | `CustomEvent<CheckboxEventDetail<any>>` |
| `b2b-focus`  | Emits whenever the checkbox receives focus.                | `CustomEvent<FocusEvent>`               |


## Dependencies

### Depends on

- [b2b-input-label](../input-label)

### Graph
```mermaid
graph TD;
  b2b-checkbox --> b2b-input-label
  style b2b-checkbox fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
