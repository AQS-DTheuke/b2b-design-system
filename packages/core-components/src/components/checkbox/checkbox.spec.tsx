import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { CheckboxComponent } from './checkbox';

it('should render the checkbox', async () => {
  const page = await newSpecPage({
    components: [CheckboxComponent],
    template: () => <b2b-checkbox label="test"></b2b-checkbox>,
  });
  expect(page.root).toMatchSnapshot();
});

it('should disable the checkbox', async () => {
  const page = await newSpecPage({
    components: [CheckboxComponent],
    template: () => <b2b-checkbox label="test" disabled={true}></b2b-checkbox>,
  });
  expect(page.root).toMatchSnapshot();
});

it('should render a hint message if a hint string is specified', async () => {
  const page = await newSpecPage({
    components: [CheckboxComponent],
    template: () => (
      <b2b-checkbox label="test" hint="this is a hint"></b2b-checkbox>
    ),
  });
  expect(page.root).toMatchSnapshot();
});

it('should apply an error style if it is marked as invalid.', async () => {
  const page = await newSpecPage({
    components: [CheckboxComponent],
    template: () => (
      <b2b-checkbox
        label="test"
        error="this is an error"
        invalid></b2b-checkbox>
    ),
  });
  expect(page.root).toMatchSnapshot();
});

it('should render an error message if an error is specified and it is marked as invalid', async () => {
  const page = await newSpecPage({
    components: [CheckboxComponent],
    template: () => (
      <b2b-checkbox
        label="test"
        error="this is an error"
        invalid></b2b-checkbox>
    ),
  });
  expect(page.root).toMatchSnapshot();
});

it('should render a hint if  invalid and disabled are true at the same time', async () => {
  const page = await newSpecPage({
    components: [CheckboxComponent],
    template: () => (
      <b2b-checkbox
        label="test"
        error="This is a test error"
        hint="this is a test hint"
        invalid
        disabled></b2b-checkbox>
    ),
  });
  expect(page.root).toMatchSnapshot();
});
