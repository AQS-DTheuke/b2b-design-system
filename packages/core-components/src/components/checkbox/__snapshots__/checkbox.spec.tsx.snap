// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should apply an error style if it is marked as invalid. 1`] = `
<b2b-checkbox>
  <mock:shadow-root>
    <div class="b2b-checkbox b2b-checkbox--error">
      <div class="b2b-checkbox-items">
        <svg class="b2b-checkbox__icon" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
          <path d="M11.798 25.082c-.341 0-.681-.13-.942-.389l-7.132-7.115a1.334 1.334 0 0 1 1.884-1.888l6.19 6.175L26.391 7.307a1.334 1.334 0 0 1 1.884 1.888L12.74 24.693c-.26.259-.601.389-.942.389z"></path>
        </svg>
        <input class="b2b-checkbox__input" type="checkbox">
        <b2b-input-label>
          test
        </b2b-input-label>
      </div>
      <span>
        this is an error
      </span>
    </div>
  </mock:shadow-root>
</b2b-checkbox>
`;

exports[`should disable the checkbox 1`] = `
<b2b-checkbox disabled="">
  <mock:shadow-root>
    <div class="b2b-checkbox b2b-checkbox--disabled">
      <div class="b2b-checkbox-items">
        <svg class="b2b-checkbox__icon" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
          <path d="M11.798 25.082c-.341 0-.681-.13-.942-.389l-7.132-7.115a1.334 1.334 0 0 1 1.884-1.888l6.19 6.175L26.391 7.307a1.334 1.334 0 0 1 1.884 1.888L12.74 24.693c-.26.259-.601.389-.942.389z"></path>
        </svg>
        <input class="b2b-checkbox__input" disabled="" type="checkbox">
        <b2b-input-label disabled="">
          test
        </b2b-input-label>
      </div>
    </div>
  </mock:shadow-root>
</b2b-checkbox>
`;

exports[`should render a hint if  invalid and disabled are true at the same time 1`] = `
<b2b-checkbox disabled="">
  <mock:shadow-root>
    <div class="b2b-checkbox b2b-checkbox--disabled">
      <div class="b2b-checkbox-items">
        <svg class="b2b-checkbox__icon" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
          <path d="M11.798 25.082c-.341 0-.681-.13-.942-.389l-7.132-7.115a1.334 1.334 0 0 1 1.884-1.888l6.19 6.175L26.391 7.307a1.334 1.334 0 0 1 1.884 1.888L12.74 24.693c-.26.259-.601.389-.942.389z"></path>
        </svg>
        <input class="b2b-checkbox__input" disabled="" type="checkbox">
        <b2b-input-label disabled="">
          test
        </b2b-input-label>
      </div>
      <span>
        this is a test hint
      </span>
    </div>
  </mock:shadow-root>
</b2b-checkbox>
`;

exports[`should render a hint message if a hint string is specified 1`] = `
<b2b-checkbox>
  <mock:shadow-root>
    <div class="b2b-checkbox">
      <div class="b2b-checkbox-items">
        <svg class="b2b-checkbox__icon" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
          <path d="M11.798 25.082c-.341 0-.681-.13-.942-.389l-7.132-7.115a1.334 1.334 0 0 1 1.884-1.888l6.19 6.175L26.391 7.307a1.334 1.334 0 0 1 1.884 1.888L12.74 24.693c-.26.259-.601.389-.942.389z"></path>
        </svg>
        <input class="b2b-checkbox__input" type="checkbox">
        <b2b-input-label>
          test
        </b2b-input-label>
      </div>
      <span>
        this is a hint
      </span>
    </div>
  </mock:shadow-root>
</b2b-checkbox>
`;

exports[`should render an error message if an error is specified and it is marked as invalid 1`] = `
<b2b-checkbox>
  <mock:shadow-root>
    <div class="b2b-checkbox b2b-checkbox--error">
      <div class="b2b-checkbox-items">
        <svg class="b2b-checkbox__icon" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
          <path d="M11.798 25.082c-.341 0-.681-.13-.942-.389l-7.132-7.115a1.334 1.334 0 0 1 1.884-1.888l6.19 6.175L26.391 7.307a1.334 1.334 0 0 1 1.884 1.888L12.74 24.693c-.26.259-.601.389-.942.389z"></path>
        </svg>
        <input class="b2b-checkbox__input" type="checkbox">
        <b2b-input-label>
          test
        </b2b-input-label>
      </div>
      <span>
        this is an error
      </span>
    </div>
  </mock:shadow-root>
</b2b-checkbox>
`;

exports[`should render the checkbox 1`] = `
<b2b-checkbox>
  <mock:shadow-root>
    <div class="b2b-checkbox">
      <div class="b2b-checkbox-items">
        <svg class="b2b-checkbox__icon" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
          <path d="M11.798 25.082c-.341 0-.681-.13-.942-.389l-7.132-7.115a1.334 1.334 0 0 1 1.884-1.888l6.19 6.175L26.391 7.307a1.334 1.334 0 0 1 1.884 1.888L12.74 24.693c-.26.259-.601.389-.942.389z"></path>
        </svg>
        <input class="b2b-checkbox__input" type="checkbox">
        <b2b-input-label>
          test
        </b2b-input-label>
      </div>
    </div>
  </mock:shadow-root>
</b2b-checkbox>
`;
